### PR TITLE
Upgrade react native to 0.59.8

### PR DESCRIPTION
**Pull request by adamczyk777**
# Motivation
Keep example app up to date with the latest version of React Native.
# Changes
Applied rn-diff-purge changes, fixed CI (was not passing after upgrade due to detox init timeouts)

### CODE REVIEW

**Comment by osdnk** [comment]
Also, please consider bumping e2e app as well

File: e2e/babel.config.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
```
Comment (adamczyk777) [inline e2e/babel.config.js]: check

File: e2e/package.json [hunk 2]
```diff
@@ -3,34 +3,34 @@
   "version": "0.0.1",
   "private": true,
   "scripts": {
-    "start": "node node_modules/react-native/local-cli/cli.js start --config ../../../../metro.config.js",
+    "start": "node node_modules/react-native/local-cli/cli.js start --config ./metro.config.js",
```
Comment (adamczyk777) [inline e2e/package.json]: check whether it's not possible to get rid of this hacky config

File: .travis.yml [hunk 3]
```diff
@@ -1,104 +1,99 @@
...
-      - $HOME/.yarn-cache
-      - $HOME/.gradle/caches/
-      - $HOME/.gradle/wrapper/
-    android:
-      components:
-      - tools
-      - platform-tools
-      - build-tools-27.0.3
+      script:
+        - cd ./android && ./gradlew assembleDebug assembleAndroidTest -DtestBuildType=debug
```
Comment (adamczyk777) [inline .travis.yml]: @osdnk Why this command has `assembleAndroidTest -DtestBuildType=debug` is it some stuff that was relevant back in the old days when detox worked with android?

Comment (adamczyk777) [reply .travis.yml]: Because I think we might want to remove it and just test if build is passing for debug app 🤔

File: Example/metro.config.js [hunk 4]
```diff
@@ -7,6 +7,7 @@ const glob = require('glob-to-regexp');
 function getBlacklist() {
   const nodeModuleDirs = [
     glob(`${path.resolve(__dirname, '..')}/node_modules/*`),
+    glob(`${path.resolve(__dirname, '..')}/e2e/*`),
```
Comment (osdnk) [inline Example/metro.config.js]: @adamczyk777 
Why is it here?

File: e2e/.babelrc [hunk 5]
```diff
@@ -1,15 +0,0 @@
-{
```
Comment (osdnk) [inline e2e/.babelrc]: Why is it removed?

File: e2e/App.js [hunk 6]
```diff
@@ -63,6 +63,8 @@ const ExampleApp = createStackNavigator(
   }
 );
 
+const Container = createAppContainer(ExampleApp);
```
Comment (osdnk) [inline e2e/App.js]: Isn't it needed in Example as well?

File: e2e/__tests__/App-test.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+/**
+ * @format
+ */
+
+import 'react-native';
```
Comment (osdnk) [inline e2e/__tests__/App-test.js]: Why is it here? Is it needed?

File: e2e/android/app/BUCK [hunk 8]
```diff
@@ -0,0 +1,55 @@
+# To learn about Buck see [Docs](https://buckbuild.com/).
+# To run your application with Buck:
+# - install Buck
+# - `npm start` - to start the packager
+# - `cd android`
```
Comment (osdnk) [inline e2e/android/app/BUCK]: is it included in the plain RN starter app?

File: e2e/android/gradle.properties [hunk 9]
```diff
@@ -10,11 +10,9 @@
 # Specifies the JVM arguments used for the daemon process.
 # The setting is particularly useful for tweaking memory settings.
 # Default value: -Xmx10248m -XX:MaxPermSize=256m
-org.gradle.jvmargs=-Xmx2048m -XX:MaxPermSize=512m -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8
+# org.gradle.jvmargs=-Xmx2048m -XX:MaxPermSize=512m -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8
```
Comment (osdnk) [inline e2e/android/gradle.properties]: hm?

File: e2e/android/settings.gradle [hunk 10]
```diff
@@ -1,9 +1,5 @@
-rootProject.name = 'RNGH Example'
-include ':react-native-vector-icons'
-project(':react-native-vector-icons').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-vector-icons/android')
+rootProject.name = 'ExampleE2E'
 
 include ':app'
 include ':react-native-gesture-handler'
 project(':react-native-gesture-handler').projectDir = new File(rootProject.projectDir, '../../android')
-include ':detox'
```
Comment (osdnk) [inline e2e/android/settings.gradle]: Why?

**Pull request merged**